### ISSUE TITLE
Is it possible to plot val and train losses on the same figure?

### ISSUE THREAD

**Issue by cruigo93** [PyTorchLightning/pytorch-lightning#2563]
I could not find the way to do it, maybe there is something I've been missing.

**Comment by s-rog** [comment]
this is a tensorboard limitation unfortunately

**Comment by awaelchli** [comment]
not possible in TB as far as I know. with some of the other loggers you can do that in the UI.

**Comment by cruigo93** [comment]
just as far as I know, it is possible in catalyst, and I found tutorial in medium where author plot cos and sin in the same pytorch+tensorboard figure
https://medium.com/@rktkek456/pytorch-tensorboard-tutorial-for-a-beginner-b037ee66574a

**Comment by awaelchli** [comment]
I see now, this is different. There they call add_scalars (instead of add_scalar) and this creates multiple experiments (see screenshots) and then it is possible to select them to overlap the plots. 
So in PL you could do 
```python
# in training step
self.logger.experiment.add_scalars("losses", {"train_loss": loss})
# in val/test step
self.logger.experiment.add_scalars("losses", {"val_loss": loss})
```
Maybe we can support this as part of the output dict of trainingstep etc. but I'm not sure what will happen since other loggers won't support this nested dict structure.

**Comment by cruigo93** [comment]
Thank you! I see now, I will try just with calling logger, so I will close this issue once there is one the same

**Comment by jovenwayfarer** [comment]
@awaelchli hi, how do the same but have epoch on x-axis, not steps?

**Comment by awaelchli** [comment]
I don't know what it will look like, but try: 
```
self.logger.experiment.add_scalars("losses", {"val_loss": loss}, global_step=self.current_epoch)
```
I am not very familiar with TB.

**Comment by mfoglio** [comment]
> I don't know what it will look like, but try:
> 
> ```
> self.logger.experiment.add_scalars("losses", {"val_loss": loss}, global_step=self.current_epoch)
> ```
> 
> I am not very familiar with TB.

I have been trying this but it is not logging anything for me.
I used to another method in the past but it's not working anymore after updating the library: https://github.com/Lightning-AI/torchmetrics/discussions/2396

**Comment by caenrigen** [comment]
For future readers, you might want to consider using `trainer.logger.experiment.add_custom_scalars_multilinechart` to display several scalars in the `CUSTOM SCALAR` tab of TensorBoard.

More details in my [SO answer](https://stackoverflow.com/a/79644264/9047715)
